sdk/src/transaction/sanitized.rs

```diff
@@ -50,18 +50,30 @@ impl AddressLoader for DisabledAddressLoader {
     }
 }
 
+/// Type that represents whether the transaction message has been precomputed or
+/// not.
+pub enum MessageHash {
+    Precomputed(Hash),
+    Compute,
+}
+
 impl SanitizedTransaction {
     /// Create a sanitized transaction from an unsanitized transaction.
     /// If the input transaction uses address tables, attempt to lookup
     /// the address for each table index.
     pub fn try_create(
         tx: VersionedTransaction,
-        message_hash: Hash,
+        message_hash: MessageHash,
```

Comment:
since this is a published, public interface, wdyt about making this `Into<MessageHash>`, then implementing that on `Hash` to avoid API breakage?

Reply:
I'm actually hoping to take this out of sdk altogether but for now using `Into<MessageHash>` sounds good. Updated.